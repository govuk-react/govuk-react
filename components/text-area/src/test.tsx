import * as React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import Textarea from '.';

describe('Textarea', () => {
  let props;
  const example = 'example';
  const meta = {
    touched: true,
    error: example,
  };
  const wrapper = <Textarea>{example}</Textarea>;
  const wrapperErrorText = <Textarea meta={meta}>{example}</Textarea>;
  const wrapperHint = <Textarea hint={example}>{example}</Textarea>;

  beforeEach(() => {
    props = {
      children: example,
    };
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('passes `props.children` to the rendered `wrapper` as `children`', () => {
    expect(wrapper.props.children).toBe(props.children);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });

  it('matches snapshot for error', () => {
    expect(mount(wrapperErrorText)).toMatchSnapshot('wrapperErrorText mount');
  });

  it('matches snapshot for hint', () => {
    expect(mount(wrapperHint)).toMatchSnapshot('wrapperHint mount');
  });
});
