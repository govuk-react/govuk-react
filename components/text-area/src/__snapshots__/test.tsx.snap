// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea matches snapshot for error: wrapperErrorText mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  border-left: 4px solid #d4351c;
  margin-right: 15px;
  padding-left: 10px;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  clear: both;
  color: #d4351c;
}

.c3 {
  box-sizing: border-box;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 1.25;
  width: 100%;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
  border: 4px solid #d4351c;
}

.c3:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3;
    width: 75%;
  }
}

<TextArea
  input={Object {}}
  meta={
    Object {
      "error": "example",
      "touched": true,
    }
  }
>
  <src__Label
    error={true}
  >
    <label
      className="c0"
    >
      <LabelText>
        <span
          className="c1"
        >
          example
        </span>
      </LabelText>
      <ErrorText>
        <span
          className="c2"
        >
          example
        </span>
      </ErrorText>
      <src__TextAreaField
        error={true}
        rows={5}
      >
        <textarea
          className="c3"
          rows={5}
        />
      </src__TextAreaField>
    </label>
  </src__Label>
</TextArea>
`;

exports[`Textarea matches snapshot for hint: wrapperHint mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  color: #6f777b;
}

.c3 {
  box-sizing: border-box;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 1.25;
  width: 100%;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c3:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 19px;
    line-height: 1.3;
    width: 75%;
  }
}

<TextArea
  hint="example"
  input={Object {}}
  meta={Object {}}
>
  <src__Label
    error={false}
  >
    <label
      className="c0"
    >
      <LabelText>
        <span
          className="c1"
        >
          example
        </span>
      </LabelText>
      <HintText>
        <span
          className="c2"
        >
          example
        </span>
      </HintText>
      <src__TextAreaField
        rows={5}
      >
        <textarea
          className="c3"
          rows={5}
        />
      </src__TextAreaField>
    </label>
  </src__Label>
</TextArea>
`;

exports[`Textarea matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c2 {
  box-sizing: border-box;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 1.25;
  width: 100%;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c2:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3;
    width: 75%;
  }
}

<TextArea
  input={Object {}}
  meta={Object {}}
>
  <src__Label
    error={false}
  >
    <label
      className="c0"
    >
      <LabelText>
        <span
          className="c1"
        >
          example
        </span>
      </LabelText>
      <src__TextAreaField
        rows={5}
      >
        <textarea
          className="c2"
          rows={5}
        />
      </src__TextAreaField>
    </label>
  </src__Label>
</TextArea>
`;
